packages/gitbook/src/middleware.ts

```diff
@@ -470,10 +477,12 @@ async function lookupSiteOrSpaceInMultiIdMode(
         throw new Error('Collection is not supported in multi-id mode');
     }
 
-    const gitbookAPI = new GitBookAPI({
+    const contextKey = decoded.claims ? hash(decoded.claims) : undefined;
```

Comment:
Are these claims going to contain some useless props (`iat`, etc)?

We don't want to bust the cache on useless things.

Also this condition will always be true for VA; causing the cache to be busted for all VA website.

```suggestion
    const contextKey = decoded.claims && Object.keys(decoded.claims).length > 0 ? hash(decoded.claims) : undefined;
```

Reply:
Also generally I don't understand why we are computing our own hash here; while at line 158 we use the one from the published content resolution?

Reply:
> Also generally I don't understand why we are computing our own hash here; while at line 158 we use the one from the published content resolution?

This part is for `lookupSiteOrSpaceInMultiIdMode` (e.g preview) which I don't think it calls `getPublishedContentByURL` (the one that returns the contextID) which is way I'm doing it manually based on the "preview" token provided in the url. 

> Are these claims going to contain some useless props (iat, etc)?
>
> We don't want to bust the cache on useless things.
> 
> Also this condition will always be true for VA; causing the cache to be busted for all VA website.

This logic applies only `lookupSiteOrSpaceInMultiIdMode` which I think doesn't handle VA.

Reply:
Ha yes it makes sense 👍 

So yeah we need a hash to ensure we can use the preview for adaptive content

Reply:
Changed it to pull the context from the URL intead of parsing it from the token claims. I'll make in parallel the changes to the getSitePublishingPreviewById API endpoint to return the preview URL with the context hash when it was provided claims.

Reply:
It's for preview this mode no? Also I know but other than pulling it from the token and computing a hash in GBO I don't see any other option.

Reply:
I think computing the hash from the token's claims is good. My original comment was missing the context of the why.